improver_tests/utilities/copy_attributes/test_CopyAttributes.py

```diff
@@ -0,0 +1,99 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+from unittest.mock import patch, sentinel
+
+from iris.cube import Cube
+
+from improver.utilities.copy_attributes import CopyAttributes
+
+
+class HaltExecution(Exception):
+    pass
+
+
+@patch("improver.utilities.copy_attributes.as_cube")
+@patch("improver.utilities.copy_attributes.as_cubelist")
+def test_as_cubelist_called(mock_as_cubelist, mock_as_cube):
+    mock_as_cubelist.return_value = sentinel.cubelist
+    mock_as_cube.side_effect = HaltExecution
+    try:
+        CopyAttributes(["attribA", "attribB"])(
+            sentinel.cube0, sentinel.cube1, template_cube=sentinel.template_cube
+        )
+    except HaltExecution:
+        pass
+    mock_as_cubelist.assert_called_once_with(sentinel.cube0, sentinel.cube1)
+    mock_as_cube.assert_called_once_with(sentinel.template_cube)
+
+
+def test_copy_attributes_multi_input():
+    """
+    Test the copy_attributes function for multiple input cubes.
+
+    Demonstrates copying attributes from the template cube to the input
+    cubes and also demonstrates the attributes on the templates cube that
+    aren't specified in the attributes list awre indeed ignored.
+    """
+    attributes = ["attribA", "attribB"]
+    cube0 = Cube([0], attributes={"attribA": "valueA", "attribB": "valueB"})
+    cube1 = Cube([0], attributes={"attribA": "valueAA", "attribB": "valueBB"})
+    template_cube = Cube(
+        [0], attributes={"attribA": "tempA", "attribB": "tempB", "attribC": "tempC"}
+    )
+
+    plugin = CopyAttributes(attributes)
+    result = plugin.process(cube0, cube1, template_cube=template_cube)
+    assert type(result) == tuple
+    for res in result:
+        assert res.attributes["attribA"] == "tempA"
+        assert res.attributes["attribB"] == "tempB"
+        assert "attribC" not in res.attributes
+    assert id(result[0]) == id(cube0)
+    assert id(result[1]) == id(cube1)
+
+
+def test_copy_attributes_single_input():
+    """
+    As per 'test_copy_attributes_multi_input' except only one input cube is provided.
+    """
+    attributes = ["attribA", "attribB"]
+    cube0 = Cube([0], attributes={"attribA": "valueA", "attribB": "valueB"})
```

Comment:
Can you please modify these values not to match the template cube so the impact of apply the plugin can be seen.
Can you please add an attribute to cube0 that is not on the template cube (attribD) to demonstrate that it is left in place and unchanged by the application of the plugin.